py2store/persisters/sqlserver_w_pyodbc.py

```diff
@@ -0,0 +1,121 @@
+import platform
+import subprocess
+from collections.abc import MutableMapping
+
+
+class SQLServerPersister(MutableMapping):
+    def __init__(self, conn_protocol='tcp', host='localhost', port='1433', db_username='SA', db_pass='Admin123x',
+                 db_name='py2store', table_name='person', primary_key='id', data_fields=('name',)):
+
+        self.__check_dependencies()
+        self._sql_server_client = pyodbc.connect('DRIVER={{ODBC Driver 17 for SQL Server}};'
+                                                 'SERVER={}:{},{};'
+                                                 'DATABASE={};'
+                                                 'UID={};'
+                                                 'PWD={}'
+                                                 .format(conn_protocol, host, port, db_name, db_username, db_pass))
+
+        self._cursor = self._sql_server_client.cursor()
+        self._table_name = table_name
+        self._primary_key = primary_key
+
+        self._select_all_query = "SELECT * from {table};".format(table=self._table_name)
+        self._insert_query = "INSERT into {table}({{attributes}}) VALUES ({{values}});".format(table=self._table_name)
+        self._select_query = "SELECT * from {table} where {primary_key}='{{value}}'".format(
+            table=self._table_name, primary_key=self._primary_key)
+        self._del_query = "DELETE from {table} where {primary_key} = {{value}};".format(table=self._table_name,
+                                                                                        primary_key=self._primary_key)
+
+    @staticmethod
+    def __check_dependencies():
+        import pkg_resources
+        installed_packages = [pkg.project_name for pkg in pkg_resources.working_set]
+        if 'pyodbc' not in installed_packages:
+            raise ModuleNotFoundError("'SQLServerPersister' depends on the module 'pyodbc' which is not installed. "
+                                      "Try installing dependency using 'pip install pyodbc'.")
+
+        # import pyodbc globally
+        global pyodbc
+        import pyodbc
+
+        if 'ubuntu' in platform.platform().lower():
+            result = subprocess.Popen(["dpkg", "-s", "msodbcsql17"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
This code will be executed every time you __init__ a new SQLServerPersister instance.

Reply:
This is the intended behaviour, As py2store will be used for every storage type, but whenever user wants to use to for the SQLServer and creates an instance of the `SQLServerPersister`, we should check for dependencies of the SQLServer.